File: PortfolioAnalyticsAPI/Data/PageVisit.cs
//Data/PageVisit.cs
namespace PortfolioAnalyticsAPI.Data;

public class PageVisit
{
    public long Id { get; set; }
    public required string Path { get; set; }
    public required string VisitorHash { get; set; }
    public DateTime Timestamp { get; set; } = DateTime.UtcNow;
}

File: backend-portofolio/internal/core/domain/page_visit.go
package domain

import "time"

type PageVisit struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	Path        string    `gorm:"type:text;not null" json:"path"`
	VisitorHash string    `gorm:"type:varchar(255);not null;index" json:"visitor_hash"`
	Timestamp   time.Time `json:"timestamp"`
}

type VisitorSummary struct {
	VisitorHash    string
	FirstVisit     string
	LastVisit      string
	TotalPageViews int64
}
